### ISSUE TITLE
Handle the case "y_train is a series while X_train is a numpy array"

### ISSUE THREAD

**Issue by sonichi** [microsoft/FLAML#777]
Python 3.9, pandas 1.5.1.

```python
from sklearn.datasets import make_classification
from flaml import AutoML
from pandas import Series


if __name__ == "__main__":

    x_train, y_train = make_classification(1279, 20)
    y_train = Series(y_train, index=range(len(y_train), 0, -1))

    automl = AutoML()
    automl.fit(X_train=x_train, y_train=y_train, X_val=x_train, y_val=y_train, task="regression", max_iter=2)
```

flaml/automl.py:290: FutureWarning: The behavior of `series[i:j]` with an integer-dtype index is deprecated. In a future version, this will be treated as *label-based* indexing, consistent with e.g. `series[i]` lookups. To retain the old behavior, use `series.iloc[i:j]`. To get the future behavior, use `series.loc[i:j]`

Solution: make the indentation of https://github.com/microsoft/FLAML/blob/b63f1bc8b8b511932d3c519e0e214bd706e94068/flaml/automl.py#L1276-L1277
one level left.

**Comment by evensure** [comment]
I'd like to make that my first "real" issue. Please assign

**Comment by evensure** [comment]
@sonichi I know the issue is rather trivial, however I wanted to reproduce the error first so that I can see if there'd be more than only line 290 with FutureWarnings related to this issue. Could you give me some guidance, please?

I followed the steps from [https://microsoft.github.io/FLAML/docs/Contribute/](https://microsoft.github.io/FLAML/docs/Contribute/)
`pip install -e FLAML[test,notebook]` didn't work:
![image](https://user-images.githubusercontent.com/42479629/199031535-50fcee27-6a70-4e05-84aa-a96eecd3083a.png)

Instead, I opened it in a container in vscode, so I should be set.

**Comment by sonichi** [comment]
> @sonichi I know the issue is rather trivial, however I wanted to reproduce the error first so that I can see if there'd be more than only line 290 with FutureWarnings related to this issue. Could you give me some guidance, please?
> 
> I followed the steps from https://microsoft.github.io/FLAML/docs/Contribute/ `pip install -e FLAML[test,notebook]` didn't work: ![image](https://user-images.githubusercontent.com/42479629/199031535-50fcee27-6a70-4e05-84aa-a96eecd3083a.png)
> 
> Instead, I opened it in a container in vscode, so I should be set.

Sure. Thanks for working on it. First, about the pip install error: did you run the command after git clone? "FLAML" is supposed to be a relative path you cloned into. If you clone into a different path, replace "FLAML" with that path.
I'll update the issue content about how to reproduce this warning.

**Comment by evensure** [comment]
Sure, glad I can help... hopefully :D

I did run it right after! I just spotted my the error, needed to escape the brackets!

**Comment by sonichi** [comment]
> Sure, glad I can help... hopefully :D
> 
> I did run it right after! I just spotted my the error, needed to escape the brackets!

I see. Maybe you can create a PR to update the documentation website first, about the escaping. :)

**Comment by sonichi** [comment]
@evensure After some investigation, I found that this issue only appears when X_train is a numpy array and y_train is a pandas series.
And the solution is very simple. Please see the updated issue.

**Comment by evensure** [comment]
@sonichi Is there an advantage of your solution against dealing with it in the same manner as for `X_train` in the lines before?

https://github.com/microsoft/FLAML/blob/b63f1bc8b8b511932d3c519e0e214bd706e94068/flaml/automl.py#L286-L289

I'd have proposed this in order to omit potential side-effects from `.reset_index()`:
```python
if isinstance(self.y_train, pd.Series):
    sampled_y_train = self.y_train.iloc[:sample_size]
else:
    sampled_y_train = self.y_train[:sample_size]
```

**Comment by sonichi** [comment]
> @sonichi Is there an advantage of your solution against dealing with it in the same manner as for `X_train` in the lines before?
> 
> https://github.com/microsoft/FLAML/blob/b63f1bc8b8b511932d3c519e0e214bd706e94068/flaml/automl.py#L286-L289
> 
> I'd have proposed this in order to omit potential side-effects from `.reset_index()`:
> 
> ```python
> if isinstance(self.y_train, pd.Series):
>     sampled_y_train = self.y_train.iloc[:sample_size]
> else:
>     sampled_y_train = self.y_train[:sample_size]
> ```

The reason is the shuffling: https://github.com/microsoft/FLAML/blob/b63f1bc8b8b511932d3c519e0e214bd706e94068/flaml/automl.py#L1259-L1264
Maybe it works without resetting the index if you use .iloc. But then we need to search for all the places using [:] and replace with .iloc.
BTW, we should check `self._state.sample_weight_all` and either reset index or use .iloc too.
